.github/workflows/test.yml

```diff
@@ -4,101 +4,165 @@ name: Test
+      
       - name: Install Rust Toolchain
         uses: RDXWorks-actions/toolchain@master
         with:
-          toolchain: nightly-2024-01-11
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
           default: true
-          components: rustfmt, clippy
-      - name: Install cargo nextest
-        uses: RDXWorks-actions/cargo-install@main
-        with:
-          crate: cargo-nextest
-          locked: true
-      - name: Install JNA
-        run: wget https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar
-      - name: Run sccache-cache
-        uses: RDXWorks-actions/sccache-action@main
+            
       - name: Run tests
-        run: CLASSPATH="$PWD/jna-5.13.0.jar" cargo nextest run
+        run: cargo test
+
+# Swift testing on macOS (Apple Silicon)
+  test-swift:
+    runs-on: macos-13-xlarge
+    needs: check-clippy
+    steps:
+      - uses: RDXWorks-actions/checkout@main
+
+      - name: Install Rust Toolchain for aarch64-apple-darwin
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          target: aarch64-apple-darwin
+
+      - name: Build for macOS target and run swift test
         env:
-          # Required env-var to increase the heapsize when using `kotlinc`.
-          JAVA_OPTS: "-Xmx8g"
-          # Enable sccache
-          SCCACHE_GHA_ENABLED: "true"
-          RUSTC_WRAPPER: "sccache"
-  codecoverage:
-    runs-on: ubuntu-latest
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          rustup target add aarch64-apple-darwin
+          sh ./scripts/ios/test.sh
+
+# Kotlin test run on JVM on macOS (Apple Silicon)
+  test-kotlin:
+    runs-on: macos-13-xlarge
+    needs: check-clippy
     steps:
       - uses: RDXWorks-actions/checkout@main
+
+      - name: Install Rust Toolchain for aarch64-linux-android
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          components: ${{ env.RUST_COMPONENTS }}
+          target: aarch64-linux-android
+        
+      - name: Install Rust Toolchain for armv7-linux-androideabi
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          components: ${{ env.RUST_COMPONENTS }}
+          target: armv7-linux-androideabi
+
+      - name: Install Rust Toolchain for aarch64-apple-darwin
+        uses: RDXWorks-actions/toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          target: aarch64-apple-darwin
+
+      - name: Set up JDK 17
+        uses: RDXWorks-actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+      
+      - name: Setup Android SDK
+        uses: RDXWorks-actions/setup-android@v2
+      
+      - name: Install cargo-ndk
+        run: cargo install cargo-ndk
+  
+      - name: Test JVM
+        run: ./jvm/gradlew -p jvm/sargon-android testDebug
```

Comment:
Change to `./jvm/gradlew -p jvm/sargon-android testDebugUnitTest` since only unit tests are itegrated.